crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs

```diff
@@ -48,21 +50,51 @@ pub struct NoRestrictedImportsConfig {
 #[serde(rename_all = "camelCase")]
 struct RestrictedPath {
     name: CompactStr,
-    import_names: Option<Box<[CompactStr]>>,
-    allow_import_names: Option<Box<[CompactStr]>>,
+    import_names: Option<Vec<CompactStr>>,
+    allow_import_names: Option<Vec<CompactStr>>,
     message: Option<CompactStr>,
 }
 
 #[derive(Debug, Clone, Deserialize)]
 #[serde(rename_all = "camelCase")]
 struct RestrictedPattern {
-    group: Vec<CompactStr>,
-    import_names: Option<Box<[CompactStr]>>,
-    allow_import_names: Option<Box<[CompactStr]>>,
+    group: Option<Vec<CompactStr>>,
+    regex: Option<SerdeRegexWrapper<Regex>>,
+    import_names: Option<Vec<CompactStr>>,
+    import_name_pattern: Option<SerdeRegexWrapper<Regex>>,
+    allow_import_names: Option<Vec<CompactStr>>,
+    allow_import_name_pattern: Option<SerdeRegexWrapper<Regex>>,
     case_sensitive: Option<bool>,
     message: Option<CompactStr>,
 }
 
+/// A wrapper type which implements `Serialize` and `Deserialize` for
+/// types involving `Regex`
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+pub struct SerdeRegexWrapper<T>(pub T);
+
+impl std::ops::Deref for SerdeRegexWrapper<Regex> {
+    type Target = Regex;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<'de> Deserialize<'de> for SerdeRegexWrapper<Regex> {
+    fn deserialize<D>(d: D) -> Result<SerdeRegexWrapper<Regex>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = <Cow<str>>::deserialize(d)?;
+
+        match s.parse() {
+            Ok(regex) => Ok(SerdeRegexWrapper(regex)),
+            Err(err) => Err(D::Error::custom(err)),
+        }
```

Comment:
Just a future note for performance optimization: parsing regex is generally pretty slow, so we try to do it as little as possible. It might even be worth refactoring this in the future to recognize common patterns such as `^foo` and `bar$` and rewrite them to use `starts_with()` and `ends_with()` instead which are just as fast and don't require any allocation beforehand

Reply:
We need to check for altneratives then. like `^foo|^bar`.
Someone in the future can try to optimize it :)